applications/DG-Max/Programs/DGMaxEigenConvergence.cpp

```diff
@@ -0,0 +1,205 @@
+
+ 2. Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following disclaimer in the documentation
+ and/or other materials provided with the distribution.
+
+ 3. Neither the name of the copyright holder nor the names of its contributors
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// Program to run convergence tests
+
+#include <iomanip>
+#include <utility>
+#include "Base/CommandLineOptions.h"
+
+#include "DGMaxLogger.h"
+#include "DGMaxProgramUtils.h"
+#include "Algorithms/DivDGMaxEigenvalue.h"
+#include "Algorithms/DGMaxEigenvalue.h"
+#include "Utils/HomogeneousBandStructure.h"
+#include "Utils/Verification/AbstractEVConvergenceTest.h"
+#include "Utils/Verification/DGMaxEVConvergenceTest.h"
+#include "Utils/Verification/DivDGMaxEVConvergenceTest.h"
+#include "Utils/Verification/EVConvergenceResult.h"
+
+auto &dimensionArg = Base::register_argument<std::size_t>(
+    'd', "dimension", "The dimension of the problem", true);
+
+auto &structureArg = Base::register_argument<std::size_t>(
+    '\0', "structure", "The structure to use", true, 0);
+
+auto &meshFiles = Base::register_argument<std::string>(
+    'm', "meshes", "The mesh files to use, comma separated", true);
+
+auto &method = Base::register_argument<std::string>(
+    '\0', "method",
+    "The method to be used, either 'DGMAX' or 'DIVDGMAX' (default)", false,
+    "DIVDGMAX");
+
+/// Create a reference bandstructure based on the structure index
+template <std::size_t DIM>
+std::unique_ptr<BandStructure<DIM>> createStructure(
+    std::size_t structureIndex) {
+    // Using same structure indices as for the Jelmer structure
+    if (structureIndex == 0) {
+        std::array<LinearAlgebra::SmallVector<DIM>, DIM> reciprocalVectors;
+        for (std::size_t i = 0; i < DIM; ++i) {
+            for (std::size_t j = 0; j < DIM; ++j) {
+                reciprocalVectors[i][j] = 0;
+            }
+            // Assuming unit cube as unit cell
+            reciprocalVectors[i][i] = 2 * M_PI;
+        }
+        return std::unique_ptr<BandStructure<DIM>>(
```

Comment:
preferably make_unique, there is also a clang-tidy workflow for it.

Reply:
I have fixed these now, but unfortunately I'm still messing with getting the clang-tidy options into clion. Do you think it would be worth it if we create a .clang-tidy for the project and enforce it?